File: <100/006/006.py
def main() -> None:
    square_sum = sum((j**2 for j in range(1, 101)))
    sum_square = sum((j for j in range(1, 101))) ** 2
    answer = sum_square - square_sum

    print(answer)

    return


if __name__ == '__main__':
    main()

File: <100/006/006.hs
main :: IO ()
main = print $ diffSquares start terminal


diffSquares :: (Num a, Enum a) => a -> a -> a
diffSquares s t = squareSum s t - sumSquares s t


squareSum :: (Num a, Enum a) => a -> a -> a
squareSum  s t = square . sum $ [s..t]


sumSquares :: (Num a, Enum a) => a -> a -> a
sumSquares s t = sum [square x | x <- [s..t]]


square:: Num a => a -> a
square x = x * x


----------------------- supplemental -----------------------
start :: Int
start = 1


terminal :: Int
terminal =  100
